packages/core/upload/server/src/services/upload.ts

```diff
@@ -553,9 +553,48 @@ export default ({ strapi }: { strapi: Core.Strapi }) => {
     });
   }
 
+  /**
+   * Uploads a file by path, buffer or stream programmaticaly
+   * @public
+   */
+  async function uploadFile(file: string | Buffer | fs.ReadStream, options: FileInfo) {
+    let filepath: string;
+
+    if (Buffer.isBuffer(file)) {
+      filepath = path.resolve(os.tmpdir(), crypto.randomBytes(10).toString('hex'));
+      await fse.writeFile(filepath, file);
+    } else if (file instanceof fs.ReadStream) {
+      filepath = path.resolve(os.tmpdir(), crypto.randomBytes(10).toString('hex'));
+      await fse.createWriteStream(filepath).write(file);
```

Comment:
On my phone rn so can't test it thoroughly but iirc you need to wrap streams in Pipeline() to prevent mem leaks. (May FSE does this under the hood)

Reply:
actually the way to avoid leaks is to store the stream on disk and then pass a path to the pipelining. this is what we recently did everywhere and it fixed both meme leaks & perf issues